packages/api/src/graphql/comment/comment.private-mutation.ts

```diff
@@ -19,3 +25,85 @@ export const takeActionOnComment = (
     }
   })
 }
+
+export const updateComment = async (
+  commentId: string,
+  text: string | undefined,
+  tagIds: string[] | undefined,
+  authenticate: Context['authenticate'],
+  commentClient: PrismaClient['comment']
+) => {
+  const {roles} = authenticate()
+  authorise(CanUpdateComments, roles)
+
+  return commentClient.update({
+    where: {id: commentId},
+    data: {
+      revisions: text
+        ? {
+            create: {
+              text
+            }
+          }
+        : undefined,
+      tags: tagIds
+        ? {
+            connectOrCreate: tagIds.map(tagId => ({
```

Comment:
What if, CAN_CREATE_TAG permission is missing?

Reply:
This is just for the m to m table, so it adds a record to bind a comment to a tag.